### PR TITLE
Dreamview: add signal and stopsign for offlineview

### PR DESCRIPTION
**Pull request by kiboliu**

### CODE REVIEW

File: modules/dreamview/frontend/src/renderer/tileground.js [hunk 1]
```diff
@@ -58,15 +80,44 @@ export default class TileGround {
...
+                    mesh.position.set(itemPos.x, itemPos.y, itemPos.z);
+                    if (item.id) {
+                        // store signal id as name in mesh
+                        mesh.name = item.id;
+                    }
+                    mesh.matrixAutoUpdate = false;
+                    mesh.updateMatrix();
+
+                    // a tile may have multiple signal or stopsign
+                    currentItems[`${key}_${item.x}_${item.y}`] = mesh;
```
Comment (vlin17) [inline modules/dreamview/frontend/src/renderer/tileground.js]: How come we want to include x,y? Key is not unique?

File: modules/dreamview/frontend/src/renderer/tileground.js [hunk 2]
```diff
@@ -22,6 +40,8 @@ export default class TileGround {
 
         this.hash = -1;
         this.currentTiles = {};
+        this.currentSignal = {};
+        this.currentStopsign = {};
```
Comment (vlin17) [inline modules/dreamview/frontend/src/renderer/tileground.js]: currentStopSign

File: modules/dreamview/frontend/src/renderer/tileground.js [hunk 3]
```diff
@@ -1,7 +1,25 @@
...
+    y: TRAFFIC_LIGHT_SCALE,
+    z: TRAFFIC_LIGHT_SCALE
+};
+
+const STOP_SIGN_SCALE = 0.01;
+const stopSignScales = {
+    x: STOP_SIGN_SCALE,
+    y: STOP_SIGN_SCALE,
+    z: STOP_SIGN_SCALE
+};
```
Comment (vlin17) [inline modules/dreamview/frontend/src/renderer/tileground.js]: These constants are used in map.js as well, is it possible to consolidate them?

File: modules/dreamview/frontend/src/renderer/tileground.js (hunk 1 above)
Comment (kiboliu) [reply modules/dreamview/frontend/src/renderer/tileground.js]: For same row and col, multiple signals can exists, thus add x,y for key.

**Pull request merged**